src/commands/clone.js

```diff
@@ -41,10 +42,37 @@ function clone (opts) {
   var onExit = require('../lib/exit')
   var parseArgs = require('../parse-args')
   var debug = require('debug')('dat')
-
   var parsed = parseArgs(opts)
+
   opts.key = parsed.key || opts._[0] // pass other links to resolver
   opts.dir = parsed.dir
+
+  // Resolves dat.json path and parses dat url from it.
+  if (fs.existsSync(opts.key)) {
+    var datPath
```

Comment:
Can we set this to `opts.key` here and use it throughout to make it a bit easier to read?

```js
var datPath = opts.key
...
if (path.basename(datPath)) {} 
```